dataprofiler/data_readers/data_utils.py

```diff
@@ -843,3 +846,99 @@ def url_to_bytes(url_as_string: Url, options: Dict) -> BytesIO:
 
     stream.seek(0)
     return stream
+
+
+def is_s3_uri(path: str) -> bool:
```

Comment:
This functionality related to s3 interactions is specific to one type of data path interacting with the profiler.
To that regard we should probably put all this functionality in a class called something like `s3_helper` and have public and private functions when applicable. This would make the functions easier to group and use IMO.
Doesnt have to be in this PR as this is sort of nitpicky issue but it would make the code make a little more sense within this `data_utils.py` file

Reply:
+1

Reply:
Thank you for your comment, The refactoring of S3_connection helper functions has been implemented, as described in previous comment.